packages/contractkit/src/identity/metadata.test.ts

```diff
@@ -4,31 +4,64 @@ import { NativeSigner } from '@celo/utils/lib/signatureUtils'
 import { newKitFromWeb3 } from '../kit'
 import { createNameClaim } from './claims/claim'
 import { ClaimTypes, IdentityMetadataWrapper } from './metadata'
-
 testWithGanache('Metadata', (web3) => {
   const kit = newKitFromWeb3(web3)
   const address = ACCOUNT_ADDRESSES[0]
   const otherAddress = ACCOUNT_ADDRESSES[1]
 
-  test('correctly recovers the claims', async () => {
+  test('correctly recovers the claims when signed by the account', async () => {
     const name = 'Celo'
     const metadata = IdentityMetadataWrapper.fromEmpty(address)
     await metadata.addClaim(createNameClaim(name), NativeSigner(kit.web3.eth.sign, address))
     const serializedMetadata = metadata.toString()
-    const parsedMetadata = IdentityMetadataWrapper.fromRawString(serializedMetadata)
+    const parsedMetadata = await IdentityMetadataWrapper.fromRawString(serializedMetadata, kit)
     const nameClaim = parsedMetadata.findClaim(ClaimTypes.NAME)
 
     expect(nameClaim).not.toBeUndefined()
     expect(nameClaim!.name).toEqual(name)
   })
 
+  test("correctly recovers the claims when the metadata is signed by any of the account's authorized signers", async () => {
+    const name = 'Celo'
+    const voteMetadata = IdentityMetadataWrapper.fromEmpty(address)
+    const validatorMetadata = IdentityMetadataWrapper.fromEmpty(address)
+    const attestationMetadata = IdentityMetadataWrapper.fromEmpty(address)
+    const accounts = await kit.contracts.getAccounts()
+    const voteSigner = ACCOUNT_ADDRESSES[2]
+    const validatorSigner = ACCOUNT_ADDRESSES[3]
+    const attestationSigner = ACCOUNT_ADDRESSES[4]
+    await accounts.createAccount().send({ from: address })
+    const testSigner = async (signer: any, action: any, metadata: any) => {
+      const pop = await accounts.generateProofOfKeyPossession(address, signer)
+      if (action === 'vote') {
```

Comment:
Optional: You could create a function map and then just call it that way. I.e.

```typescript
const functionByAction = { 'vote': acounts.authorizeVoteSigner }
await (await functionByAction[action]).send({ from: signer }))
```

Reply:
I tried something like this initially, my version was just via passing the fn as an argument, i.e. rather than passing `vote` I passed `accounts.authorizeVoteSigner`, but I can't get that or this suggestion to work - it complains about `kit` being `undefined` or similar issues, so I've stopped trying now since it seems low priority. let me know if you know what may be the problem and an easy fix